test/sm2crypttest.c

```diff
@@ -0,0 +1,247 @@
+                    const char* ctext_hex) {
+   const size_t msg_len = strlen(message);
+
+   BIGNUM* priv = NULL;
+   EC_KEY *key = NULL;
+   EC_POINT *pt = NULL;
+   unsigned char *expected = OPENSSL_hexstr2buf(ctext_hex, NULL);
+
+   size_t ctext_len = 0;
+   uint8_t *ctext = NULL;
+   uint8_t *recovered = NULL;
+   size_t recovered_len = msg_len;
+
+   int rc = 0;
+
+   BN_hex2bn(&priv, privkey_hex);
+
+   key = EC_KEY_new();
+   EC_KEY_set_group(key, group);
+   EC_KEY_set_private_key(key, priv);
+
+   pt = EC_POINT_new(group);
+   EC_POINT_mul(group, pt, priv, NULL, NULL, NULL);
+
+   EC_KEY_set_public_key(key, pt);
+   BN_free(priv);
+   EC_POINT_free(pt);
+
+   ctext_len = SM2_ciphertext_size(key, digest, msg_len);
+   ctext = OPENSSL_zalloc(ctext_len);
+   if (ctext == NULL)
+      goto done;
+
+   start_fake_rand(k_hex);
+   rc = SM2_encrypt(key, digest,
+                    (const uint8_t*)message, msg_len,
+                    ctext, &ctext_len);
+   restore_rand();
+
+   TEST_mem_eq(ctext, ctext_len, expected, ctext_len);
+   if(rc == 0)
+      goto done;
+
+   recovered = OPENSSL_zalloc(msg_len);
+   if (recovered == NULL)
+      goto done;
+   rc = SM2_decrypt(key, digest, ctext, ctext_len, recovered, &recovered_len);
+
+   TEST_int_eq(recovered_len, msg_len);
+   TEST_mem_eq(recovered, recovered_len, message, msg_len);
+   if(rc == 0)
+      return 0;
+
+   rc = 1;
+   done:
+
+   OPENSSL_free(ctext);
+   OPENSSL_free(recovered);
+   OPENSSL_free(expected);
+   EC_KEY_free(key);
+   return rc;
+   }
+
+static int sm2_crypt_test(void)
+{
+    EC_GROUP* test_group = create_EC_group(
+                "8542D69E4C044F18E8B92435BF6FF7DE457283915C45517D722EDB8B08F1DFC3",
+                "787968B4FA32C3FD2417842E73BBFEFF2F3C848B6831D7E0EC65228B3937E498",
+                "63E4C6D3B23B0C849CF84241484BFE48F61D59A5B16BA06E6E12D1DA27C5249A",
+                "421DEBD61B62EAB6746434EBC3CC315E32220B3BADD50BDC4C4E6C147FEDD43D",
+                "0680512BCBB42C07D47349D2153B70C4E5D7FDFCBFA36EA1A85841B9E46E09A2",
+                "8542D69E4C044F18E8B92435BF6FF7DD297720630485628D5AE74EE7C32E79B7",
+                "1");
+
+    if(test_group == NULL)
+       return 0;
+
+    int rc = test_sm2(test_group,
+                      EVP_sm3(),
+                      "1649AB77A00637BD5E2EFE283FBF353534AA7F7CB89463F208DDBC2920BB0DA0",
+                      "encryption standard",
+                      "004C62EEFD6ECFC2B95B92FD6C3D9575148AFA17425546D49018E5388D49DD7B4F",
+                      "307B0220245C26FB68B1DDDDB12C4B6BF9F2B6D5FE60A383B0D18D1C4144ABF1"
+                      "7F6252E7022076CB9264C2A7E88E52B19903FDC47378F605E36811F5C07423A2"
+                      "4B84400F01B804209C3D7360C30156FAB7C80A0276712DA9D8094A634B766D3A"
+                      "285E07480653426D0413650053A89B41C418B0C3AAD00D886C00286467");
+
+    if(rc == 0)
+       return 0;
+
+    // Same test as above except using SHA-256 instead of SM3
```

Comment:
Seems this comment style should be `/* */`

Reply:
Ah, yes, good catch